include/rocksdb/compaction_filter.h

```diff
@@ -158,7 +165,8 @@ class CompactionFilter {
                             const Slice& existing_value, std::string* new_value,
                             std::string* /*skip_until*/) const {
     switch (value_type) {
-      case ValueType::kValue: {
+      case ValueType::kValue:
+      case ValueType::kBlobIndex: {
```

Comment:
The reason we need to add ValueType::kBlobIndex is to support scenario when user only provide overridden Filter() api not FilterV2(). See the implementation for BlobIndexCompactionFilterBase::FilterV2() line:94, we only call ucf->FilterV2(), so we need this to wrapper to redirect the call to Filter() in user compaction filter. 
But we can also pass kValue into ucf->FilterV2() instead of value_type, since user may not expect internal concept kBlobIndex. Let me try that approach.

Reply:
Oh yeah, we should definitely pass `kValue` even if the value originally came from a blob file.